src/Microsoft.ML.Transforms/CustomMappingFactory.cs

```diff
@@ -43,4 +43,34 @@ ITransformer ICustomMappingFactory.CreateTransformer(IHostEnvironment env, strin
             return new CustomMappingTransformer<TSrc, TDst>(env, mapAction, contractName);
         }
     }
+
+    /// <summary>
+    /// The base type for stateful custom mapping factories.
+    /// </summary>
+    /// <typeparam name="TSrc">The type that describes what 'source' columns are consumed from the input <see cref="IDataView"/>.</typeparam>
+    /// <typeparam name="TState">The type that describes the state object the mapping uses.</typeparam>
+    /// <typeparam name="TDst">The type that describes what new columns are added by this transform.</typeparam>
+    public abstract class StatefulCustomMappingFactory<TSrc, TState, TDst> : ICustomMappingFactory
+        where TSrc : class, new()
```

Comment:
Tiny nit: CustomMappingFactory uses <TSrc, TDst>. Would it be more consistent to order these as <TSrc, TDst, TState> or <TState, TSrc, TDst> ?